Sources/Bot/Services/DispatchService.swift

```diff
@@ -0,0 +1,198 @@
+        if let service = mergeServices[targetBranch] {
+            service.pullRequestChangesObserver.send(value: (event.pullRequestMetadata, event.action))
+        } else {
+            makeMergeService(targetBranch: targetBranch, scheduler: self.scheduler, initialPullRequests: [event.pullRequestMetadata.reference])
+        }
+    }
+
+    private func statusChecksDidChange(event: StatusEvent) {
+        // No way to know which MergeService this event is supposed to be for – isRelative(toBranch:) only checks for head branch not target so not useful here
+        // So we're sending it to all MergeServices, and they'll filter them themselves based on their own queues
+        for mergeServiceForBranch in mergeServices.values {
+            mergeServiceForBranch.statusChecksCompletionObserver.send(value: event)
+        }
+    }
+
+    @discardableResult
+    private func makeMergeService(targetBranch: String, scheduler: DateScheduler, initialPullRequests: [PullRequest] = []) -> MergeService {
+        let mergeService = MergeService(
+            targetBranch: targetBranch,
+            integrationLabel: integrationLabel,
+            topPriorityLabels: topPriorityLabels,
+            requiresAllStatusChecks: requiresAllStatusChecks,
+            statusChecksTimeout: statusChecksTimeout,
+            initialPullRequests: initialPullRequests,
+            logger: logger,
+            gitHubAPI: gitHubAPI,
+            scheduler: scheduler
+        )
+        self.mergeServices[targetBranch] = mergeService
+        self.healthcheck.startMonitoring(mergeServiceHealthcheck: mergeService.healthcheck)
+        mergeServiceLifecycleObserver?.send(value: .created(mergeService))
+        mergeService.state.producer
+            .observe(on: scheduler)
+            .startWithValues { [weak self, service = mergeService] state in
+                self?.mergeServiceLifecycleObserver?.send(value: .stateChanged(service))
+                if state.status == .idle {
+                    self?.mergeServices[targetBranch] = nil
+                    self?.mergeServiceLifecycleObserver?.send(value: .destroyed(service))
+                }
+            }
+
+        return mergeService
+    }
+}
+
+extension DispatchService {
+    public enum MergeServiceLifecycleEvent {
+        case created(MergeService)
+        case destroyed(MergeService)
+        case stateChanged(MergeService)
+    }
+}
+
+extension DispatchService {
+    public var queuesDescription: String {
+        guard !mergeServices.isEmpty else {
+            return "No PR pending, all queues empty."
+        }
+        return mergeServices.map { (entry: (key: String, value: MergeService)) -> String in
+            """
+            ## Merge Queue for target branch: \(entry.key) ##
+
+            \(entry.value.state.value)
+            """
+        }.joined(separator: "\n\n")
+    }
+
+    public var queueStates: [MergeService.State] {
+        return self.mergeServices.values.map { $0.state.value }
```

Comment:
should we maybe define coding keys for state manually so that we don't encode things from the state that we don't need, like labels?

Reply:
Well I though you defined those manually in your PR about JSON output already? 🤔 

But you're right, I forgot to add the `targetBranch` as part of the keys since you customized the CodingKeys in your PR (see lines 768-779 of `MergeService.swift`), I should add this new `targetBranch` key indeed.

Reply:
I guess we are both right)

Reply:
Fixed + added relevant unit tests in 309bdfecf3d3245fb15abc60b96540d9f0561baf